<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sebastian Ayala Ruano">
<meta name="dcterms.date" content="2022-11-28">

<title>R programming skills summary - Experimental Design and Data Management 2022 - 2&nbsp; Week4 - Introduction to Gene expression analysis part 2</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./MSB1005_SkillsTrainingWeek3_2022.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Week4 - Introduction to Gene expression analysis part 2</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">R programming skills summary - Experimental Design and Data Management 2022</a> 
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Preface</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./MSB1005_SkillsTrainingWeek3_2022.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Week3 - Introduction to Gene expression analysis part 1</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./MSB1005_SkillsTrainingWeek4_2022.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Week4 - Introduction to Gene expression analysis part 2</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#assignment-3-data-exploration-on-the-gene-level" id="toc-assignment-3-data-exploration-on-the-gene-level" class="nav-link active" data-scroll-target="#assignment-3-data-exploration-on-the-gene-level"><span class="toc-section-number">2.1</span>  Assignment 3: Data exploration on the gene level</a>
  <ul class="collapse">
  <li><a href="#convert-the-cpm-values-to-fpkm-values.-for-some-of-the-exercises-below-we-need-to-convert-the-cpm-expression-values-to-fpkm-expression-values." id="toc-convert-the-cpm-values-to-fpkm-values.-for-some-of-the-exercises-below-we-need-to-convert-the-cpm-expression-values-to-fpkm-expression-values." class="nav-link" data-scroll-target="#convert-the-cpm-values-to-fpkm-values.-for-some-of-the-exercises-below-we-need-to-convert-the-cpm-expression-values-to-fpkm-expression-values."><span class="toc-section-number">2.1.1</span>  Convert the CPM values to FPKM values. For some of the exercises below, we need to convert the CPM expression values to FPKM expression values.</a></li>
  <li><a href="#what-does-fpkm-stand-for-how-does-this-measure-differ-from-cpm-google" id="toc-what-does-fpkm-stand-for-how-does-this-measure-differ-from-cpm-google" class="nav-link" data-scroll-target="#what-does-fpkm-stand-for-how-does-this-measure-differ-from-cpm-google"><span class="toc-section-number">2.1.2</span>  What does FPKM stand for? How does this measure differ from CPM? (Google)</a></li>
  <li><a href="#in-you-own-words-describe-what-the-code-above-does." id="toc-in-you-own-words-describe-what-the-code-above-does." class="nav-link" data-scroll-target="#in-you-own-words-describe-what-the-code-above-does."><span class="toc-section-number">2.1.3</span>  In you own words, describe what the code above does.</a></li>
  <li><a href="#can-we-compare-the-fpkm-value-of-gene-a-between-two-samples-to-state-in-which-sample-gene-a-is-more-highly-expressed" id="toc-can-we-compare-the-fpkm-value-of-gene-a-between-two-samples-to-state-in-which-sample-gene-a-is-more-highly-expressed" class="nav-link" data-scroll-target="#can-we-compare-the-fpkm-value-of-gene-a-between-two-samples-to-state-in-which-sample-gene-a-is-more-highly-expressed"><span class="toc-section-number">2.1.4</span>  Can we compare the FPKM value of gene A between two samples to state in which sample gene A is more highly expressed?</a></li>
  <li><a href="#can-we-compare-the-fpkm-value-of-gene-a-to-gene-b-in-a-single-sample-to-state-which-gene-is-more-highly-expressed" id="toc-can-we-compare-the-fpkm-value-of-gene-a-to-gene-b-in-a-single-sample-to-state-which-gene-is-more-highly-expressed" class="nav-link" data-scroll-target="#can-we-compare-the-fpkm-value-of-gene-a-to-gene-b-in-a-single-sample-to-state-which-gene-is-more-highly-expressed"><span class="toc-section-number">2.1.5</span>  Can we compare the FPKM value of gene A to gene B in a single sample to state which gene is more highly expressed?</a></li>
  <li><a href="#can-we-compare-the-cpm-values-of-gene-a-between-two-samples-to-determine-in-which-sample-gene-a-is-more-highly-expressed" id="toc-can-we-compare-the-cpm-values-of-gene-a-between-two-samples-to-determine-in-which-sample-gene-a-is-more-highly-expressed" class="nav-link" data-scroll-target="#can-we-compare-the-cpm-values-of-gene-a-between-two-samples-to-determine-in-which-sample-gene-a-is-more-highly-expressed"><span class="toc-section-number">2.1.6</span>  Can we compare the CPM values of gene A between two samples to determine in which sample gene A is more highly expressed?</a></li>
  <li><a href="#can-we-compare-the-cpm-value-of-gene-a-to-the-value-of-gene-b-in-a-single-sample-to-determine-which-gene-is-more-highly-expressed" id="toc-can-we-compare-the-cpm-value-of-gene-a-to-the-value-of-gene-b-in-a-single-sample-to-determine-which-gene-is-more-highly-expressed" class="nav-link" data-scroll-target="#can-we-compare-the-cpm-value-of-gene-a-to-the-value-of-gene-b-in-a-single-sample-to-determine-which-gene-is-more-highly-expressed"><span class="toc-section-number">2.1.7</span>  Can we compare the CPM value of gene A to the value of gene B in a single sample to determine which gene is more highly expressed?</a></li>
  <li><a href="#using-the-fpkm-values-answer-the-following-questions" id="toc-using-the-fpkm-values-answer-the-following-questions" class="nav-link" data-scroll-target="#using-the-fpkm-values-answer-the-following-questions"><span class="toc-section-number">2.1.8</span>  Using the FPKM values, answer the following questions:</a></li>
  <li><a href="#using-the-cpm-values-answer-the-following-questions" id="toc-using-the-cpm-values-answer-the-following-questions" class="nav-link" data-scroll-target="#using-the-cpm-values-answer-the-following-questions"><span class="toc-section-number">2.1.9</span>  Using the CPM values, answer the following questions:</a></li>
  </ul></li>
  <li><a href="#assignment-4-differential-gene-expression-analysis." id="toc-assignment-4-differential-gene-expression-analysis." class="nav-link" data-scroll-target="#assignment-4-differential-gene-expression-analysis."><span class="toc-section-number">2.2</span>  Assignment 4: Differential gene expression analysis.</a>
  <ul class="collapse">
  <li><a href="#what-is-differential-gene-expression-analysis-dgea-what-are-some-of-the-most-common-packages-in-r-for-dgea" id="toc-what-is-differential-gene-expression-analysis-dgea-what-are-some-of-the-most-common-packages-in-r-for-dgea" class="nav-link" data-scroll-target="#what-is-differential-gene-expression-analysis-dgea-what-are-some-of-the-most-common-packages-in-r-for-dgea"><span class="toc-section-number">2.2.1</span>  What is differential gene expression analysis (DGEA)? What are some of the most common packages in R for DGEA?</a></li>
  <li><a href="#implement-the-steps-noted-in-the-limma-guide-for-the-magnet-dataset.-start-with-a-dgea-between-dcm-patients-and-healthy-controls." id="toc-implement-the-steps-noted-in-the-limma-guide-for-the-magnet-dataset.-start-with-a-dgea-between-dcm-patients-and-healthy-controls." class="nav-link" data-scroll-target="#implement-the-steps-noted-in-the-limma-guide-for-the-magnet-dataset.-start-with-a-dgea-between-dcm-patients-and-healthy-controls."><span class="toc-section-number">2.2.2</span>  Implement the steps noted in the limma guide for the MAGNET dataset. Start with a DGEA between DCM patients and healthy controls.</a></li>
  <li><a href="#which-co-variates-should-be-taken-along-for-correction-confounding-see-the-alcohol-causes-lung-cancer-example-from-the-lecture" id="toc-which-co-variates-should-be-taken-along-for-correction-confounding-see-the-alcohol-causes-lung-cancer-example-from-the-lecture" class="nav-link" data-scroll-target="#which-co-variates-should-be-taken-along-for-correction-confounding-see-the-alcohol-causes-lung-cancer-example-from-the-lecture"><span class="toc-section-number">2.2.3</span>  Which co-variates should be taken along for correction? (confounding; see the “alcohol causes lung cancer” example from the lecture)</a></li>
  <li><a href="#copy-the-top-200-differentially-expressed-genes-to-for-a-quick-go-enrichment-analysis.-which-processes-are-changed-between-dcm-and-controls-do-these-processes-make-biological-sense-quick-literature-check" id="toc-copy-the-top-200-differentially-expressed-genes-to-for-a-quick-go-enrichment-analysis.-which-processes-are-changed-between-dcm-and-controls-do-these-processes-make-biological-sense-quick-literature-check" class="nav-link" data-scroll-target="#copy-the-top-200-differentially-expressed-genes-to-for-a-quick-go-enrichment-analysis.-which-processes-are-changed-between-dcm-and-controls-do-these-processes-make-biological-sense-quick-literature-check"><span class="toc-section-number">2.2.4</span>  Copy the top 200 differentially expressed genes to for a quick GO enrichment analysis. Which processes are changed between DCM and controls? Do these processes make biological sense? (quick literature check!)</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Week4 - Introduction to Gene expression analysis part 2</span></h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Sebastian Ayala Ruano </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 28, 2022</p>
    </div>
  </div>
    
  </div>
  

</header>

<p>This R script serves as a scaffold for adding the code required to fulfill the assignments. It includes the assignments as well as a few hints.</p>
<p>Add the necessary code and type your answers in this document for your own record.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Import libraries</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(conflicted)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrmorant)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="assignment-3-data-exploration-on-the-gene-level" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="assignment-3-data-exploration-on-the-gene-level"><span class="header-section-number">2.1</span> Assignment 3: Data exploration on the gene level</h2>
<section id="convert-the-cpm-values-to-fpkm-values.-for-some-of-the-exercises-below-we-need-to-convert-the-cpm-expression-values-to-fpkm-expression-values." class="level3" data-number="2.1.1">
<h3 data-number="2.1.1" class="anchored" data-anchor-id="convert-the-cpm-values-to-fpkm-values.-for-some-of-the-exercises-below-we-need-to-convert-the-cpm-expression-values-to-fpkm-expression-values."><span class="header-section-number">2.1.1</span> Convert the CPM values to FPKM values. For some of the exercises below, we need to convert the CPM expression values to FPKM expression values.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load data</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>geneTotExonLengths <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"Data/MAGNET_exonLengths.txt"</span>, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>gxData <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"Data/MAGNET_GeneExpressionData_CPM_19112020.txt"</span>, <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>sampleInfo <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"Data/MAGNET_SampleData_18112022.csv"</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Add gene ID as the row names</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>gxData <span class="ot">&lt;-</span> <span class="fu">column_to_rownames</span>(gxData, <span class="at">var =</span> <span class="st">"EnsemblGeneID"</span>)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>geneTotExonLengths <span class="ot">&lt;-</span> <span class="fu">column_to_rownames</span>(geneTotExonLengths, <span class="at">var =</span> <span class="st">"EnsemblGeneID"</span>)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Check that row names are the same</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(geneTotExonLengths) <span class="sc">==</span> <span class="fu">rownames</span>(gxData)) <span class="co"># TRUE (just a check)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add sample name as the row names</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>sampleInfo <span class="ot">&lt;-</span> <span class="fu">column_to_rownames</span>(sampleInfo, <span class="at">var =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert CPM expression values to FPKM</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>cpm2fpkm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>    t <span class="ot">&lt;-</span> <span class="dv">2</span><span class="sc">^</span>(x) <span class="sc">*</span> <span class="fl">1E3</span> <span class="sc">/</span> geneTotExonLengths[, <span class="dv">1</span>] <span class="co"># . before variable makes it a hidden variable</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>gxData_fpkm <span class="ot">&lt;-</span> <span class="fu">cpm2fpkm</span>(gxData)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="what-does-fpkm-stand-for-how-does-this-measure-differ-from-cpm-google" class="level3" data-number="2.1.2">
<h3 data-number="2.1.2" class="anchored" data-anchor-id="what-does-fpkm-stand-for-how-does-this-measure-differ-from-cpm-google"><span class="header-section-number">2.1.2</span> What does FPKM stand for? How does this measure differ from CPM? (Google)</h3>
<p>These metrics attempt to normalize for sequencing depth and gene length. Normalized expression units are necessary to remove technical biases in sequenced data such as depth of sequencing and gene length, and make gene expressions directly comparable within and across samples. More sequencing depth produces more read count for a gene expressed at the same level and differences in gene length generate unequal reads count for genes expressed at the same level.</p>
<p>CPM is a basic gene expression unit that normalizes only for sequencing depth (depth-normalized counts). It is biased in some applications where the gene length influences gene expression, such as RNA-seq.</p>
<span class="math display">\[\begin{equation}{}
CPM = \frac{N \;reads \;mapped \;to \;gene \times 10^{6}}{Total \;N \;of \;mapped \;reads}
\end{equation}\]</span>
<p>RPKM (reads per kilobase of transcript per million reads mapped) is a gene expression unit that measures the expression levels (mRNA abundance) of genes or transcripts. RPKM is a gene length normalized expression unit that is used for identifying the differentially expressed genes by comparing the RPKM values between different experimental conditions. Generally, the higher the RPKM of a gene, the higher the expression of that gene.</p>
<span class="math display">\[\begin{equation}{}
RPKM = \frac{N \;reads \;mapped \;to \;gene \times 10^{3} \times 10^{6}}{Total
\;N \;of \;mapped \;reads \times  gene \;length \;in \;bp}
\end{equation}\]</span>
<p>Here, 10^3 normalizes for gene length and 10^6 for sequencing depth factor.</p>
<p>FPKM (fragments per kilobase of exon per million mapped fragments) is a gene expression unit which is analogous to RPKM. FPKM is used especially for normalizing counts for paired-end RNA-seq data in which two (left and right) reads are sequenced from the same DNA fragment. Generally, the higher the FPKM of a gene, the higher the expression of that gene.</p>
<p>When we map paired-end data, both reads or only one read with high quality from a fragment can map to reference sequence. To avoid confusion or multiple counting, the fragments to which both or single read mapped are counted and represented for FPKM calculation.</p>
</section>
<section id="in-you-own-words-describe-what-the-code-above-does." class="level3" data-number="2.1.3">
<h3 data-number="2.1.3" class="anchored" data-anchor-id="in-you-own-words-describe-what-the-code-above-does."><span class="header-section-number">2.1.3</span> In you own words, describe what the code above does.</h3>
<p>The code takes the gene expression values in CPM and normalize them by the length of the genes, according to the formula expressed before.</p>
</section>
<section id="can-we-compare-the-fpkm-value-of-gene-a-between-two-samples-to-state-in-which-sample-gene-a-is-more-highly-expressed" class="level3" data-number="2.1.4">
<h3 data-number="2.1.4" class="anchored" data-anchor-id="can-we-compare-the-fpkm-value-of-gene-a-between-two-samples-to-state-in-which-sample-gene-a-is-more-highly-expressed"><span class="header-section-number">2.1.4</span> Can we compare the FPKM value of gene A between two samples to state in which sample gene A is more highly expressed?</h3>
<p>No, because FPKM values are normalized by the length of genes, which means that we cannot compare the values across different samples.</p>
</section>
<section id="can-we-compare-the-fpkm-value-of-gene-a-to-gene-b-in-a-single-sample-to-state-which-gene-is-more-highly-expressed" class="level3" data-number="2.1.5">
<h3 data-number="2.1.5" class="anchored" data-anchor-id="can-we-compare-the-fpkm-value-of-gene-a-to-gene-b-in-a-single-sample-to-state-which-gene-is-more-highly-expressed"><span class="header-section-number">2.1.5</span> Can we compare the FPKM value of gene A to gene B in a single sample to state which gene is more highly expressed?</h3>
<p>Yes, because FPKM values are normalized by the length of the genes.</p>
</section>
<section id="can-we-compare-the-cpm-values-of-gene-a-between-two-samples-to-determine-in-which-sample-gene-a-is-more-highly-expressed" class="level3" data-number="2.1.6">
<h3 data-number="2.1.6" class="anchored" data-anchor-id="can-we-compare-the-cpm-values-of-gene-a-between-two-samples-to-determine-in-which-sample-gene-a-is-more-highly-expressed"><span class="header-section-number">2.1.6</span> Can we compare the CPM values of gene A between two samples to determine in which sample gene A is more highly expressed?</h3>
<p>Yes, because CPM values are normalized by the sequencing depth (depth-normalized counts) and they do not take into account the length of genes for the normalization process.</p>
</section>
<section id="can-we-compare-the-cpm-value-of-gene-a-to-the-value-of-gene-b-in-a-single-sample-to-determine-which-gene-is-more-highly-expressed" class="level3" data-number="2.1.7">
<h3 data-number="2.1.7" class="anchored" data-anchor-id="can-we-compare-the-cpm-value-of-gene-a-to-the-value-of-gene-b-in-a-single-sample-to-determine-which-gene-is-more-highly-expressed"><span class="header-section-number">2.1.7</span> Can we compare the CPM value of gene A to the value of gene B in a single sample to determine which gene is more highly expressed?</h3>
<p>No, because CPM values are not normalized by the length of the genes.</p>
</section>
<section id="using-the-fpkm-values-answer-the-following-questions" class="level3" data-number="2.1.8">
<h3 data-number="2.1.8" class="anchored" data-anchor-id="using-the-fpkm-values-answer-the-following-questions"><span class="header-section-number">2.1.8</span> Using the FPKM values, answer the following questions:</h3>
<section id="what-are-the-ids-of-the-5-highest-expressed-genes-what-is-their-function-according-to-the-genecards-website" class="level4" data-number="2.1.8.1">
<h4 data-number="2.1.8.1" class="anchored" data-anchor-id="what-are-the-ids-of-the-5-highest-expressed-genes-what-is-their-function-according-to-the-genecards-website"><span class="header-section-number">2.1.8.1</span> What are the IDs of the 5 highest expressed genes? What is their function according to the GeneCards website?</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a column with the mean of expression values of all samples</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>exp_mean_df <span class="ot">&lt;-</span> gxData_fpkm <span class="sc">%&gt;%</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"geneID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">exp_mean =</span> <span class="fu">mean</span>(<span class="fu">c_across</span>(C00039<span class="sc">:</span>P01640))) <span class="sc">%&gt;%</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"geneID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(exp_mean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select the 5 most expressed genes</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>max5_genes_mean <span class="ot">&lt;-</span> exp_mean_df <span class="sc">%&gt;%</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="at">n=</span> <span class="dv">5</span>, exp_mean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<table class="table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th>Gene ID</th>
<th>Name</th>
<th>Function</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000198804</td>
<td>MT-CO1</td>
<td>Contributes to cytochrome-c oxidase activity</td>
</tr>
<tr class="even">
<td>ENSG00000198899</td>
<td>MT-ATP6</td>
<td>Contributes to proton-transporting ATP synthase activity</td>
</tr>
<tr class="odd">
<td>ENSG00000198938</td>
<td>MT-CO3</td>
<td>Involved in respiratory chain complex IV assembly</td>
</tr>
<tr class="even">
<td>ENSG00000198712</td>
<td>MT-CO2</td>
<td>Contributes to cytochrome-c oxidase activity</td>
</tr>
<tr class="odd">
<td>ENSG00000198886</td>
<td>MT-ND4</td>
<td>Enables NADH dehydrogenase (ubiquinone) activity</td>
</tr>
</tbody>
</table>
<p>All of the genes are involved in processes related to mitochondria activity, which make sense because the dataset has muscle samples.</p>
</section>
<section id="what-are-the-ids-of-the-5-lowest-expressed-genes-what-is-their-function-according-to-the-genecards-website" class="level4" data-number="2.1.8.2">
<h4 data-number="2.1.8.2" class="anchored" data-anchor-id="what-are-the-ids-of-the-5-lowest-expressed-genes-what-is-their-function-according-to-the-genecards-website"><span class="header-section-number">2.1.8.2</span> What are the IDs of the 5 lowest expressed genes? What is their function according to the GeneCards website?</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select the 5 lowest expressed genes</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>min5_genes_mean <span class="ot">&lt;-</span> exp_mean_df <span class="sc">%&gt;%</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_min</span>(<span class="at">n =</span> <span class="dv">5</span>, exp_mean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<table class="table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th>Gene ID</th>
<th>Name</th>
<th>Function</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000015568</td>
<td>RGPD5</td>
<td>RAN is a small GTP-binding protein of the RAS superfamily that is associated with the nuclear membrane</td>
</tr>
<tr class="even">
<td>ENSG00000162105</td>
<td>SHANK2</td>
<td>This gene encodes a protein that is a member of the Shank family of synaptic proteins that may function as molecular scaffolds in the postsynaptic density of excitatory synapses</td>
</tr>
<tr class="odd">
<td>ENSG00000267586</td>
<td>LINC00907</td>
<td>RNA Gene, and is affiliated with the lncRNA class</td>
</tr>
<tr class="even">
<td>ENSG00000215126</td>
<td>ZNG1F</td>
<td>Predicted to enable ATP binding activity</td>
</tr>
<tr class="odd">
<td>ENSG00000183914</td>
<td>DNAH2</td>
<td>Dyneins are microtubule-associated motor protein complexes</td>
</tr>
</tbody>
</table>
<p>There are pseudogenes, RNA genes, and others related to different processes (i.e.&nbsp;synaptic genes).</p>
</section>
<section id="what-are-the-ids-of-the-5-most-variable-genes-what-is-their-function-according-to-the-genecards-website" class="level4" data-number="2.1.8.3">
<h4 data-number="2.1.8.3" class="anchored" data-anchor-id="what-are-the-ids-of-the-5-most-variable-genes-what-is-their-function-according-to-the-genecards-website"><span class="header-section-number">2.1.8.3</span> What are the IDs of the 5 most variable genes? What is their function according to the GeneCards website?</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a column with the mean of expression values of all samples</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>exp_var_df <span class="ot">&lt;-</span> gxData <span class="sc">%&gt;%</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"geneID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">exp_var =</span> <span class="fu">var</span>(<span class="fu">c_across</span>(C00039<span class="sc">:</span>P01640))) <span class="sc">%&gt;%</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"geneID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(exp_var)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select the 5 most variable genes</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>max5_genes_var <span class="ot">&lt;-</span> exp_var_df <span class="sc">%&gt;%</span> </span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="at">n=</span> <span class="dv">5</span>, exp_var)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<table class="table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th>Gene ID</th>
<th>Name</th>
<th>Function</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000198692</td>
<td>EIF1AY</td>
<td>Eukaryotic Translation Initiation Factor 1A Y-Linked</td>
</tr>
<tr class="even">
<td>ENSG00000129824</td>
<td>RPS4Y1</td>
<td>Ribosomal Protein S4 Y-Linked 1</td>
</tr>
<tr class="odd">
<td>ENSG00000114374</td>
<td>USP9Y</td>
<td>Ubiquitin Specific Peptidase 9 Y-Linked</td>
</tr>
<tr class="even">
<td>ENSG00000067048</td>
<td>DDX3Y</td>
<td>DEAD-Box Helicase 3 Y-Linked</td>
</tr>
<tr class="odd">
<td>ENSG00000012817</td>
<td>KDM5D</td>
<td>Lysine Demethylase 5D - encodes a protein containing zinc finger domains</td>
</tr>
</tbody>
</table>
<p>4 of these genes are related to Y chromosome, which are absent in the female samples.</p>
<p><strong>Note: </strong>By using the FPKM dataset, we got the same genes as the 5 mots highly expressed.</p>
<section id="what-are-the-ids-of-the-5-least-variable-stable-genes-what-is-their-function-according-to-the-genecards-website" class="level5" data-number="2.1.8.3.1">
<h5 data-number="2.1.8.3.1" class="anchored" data-anchor-id="what-are-the-ids-of-the-5-least-variable-stable-genes-what-is-their-function-according-to-the-genecards-website"><span class="header-section-number">2.1.8.3.1</span> What are the IDs of the 5 least variable (= stable!) genes? What is their function according to the GeneCards website?</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select the 5 least variable genes</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>min5_genes_var <span class="ot">&lt;-</span> exp_var_df <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_min</span>(<span class="at">n=</span> <span class="dv">5</span>, exp_var)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<table class="table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th>Gene ID</th>
<th>Name</th>
<th>Function</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000136709</td>
<td>WD Repeat Domain 33</td>
<td>WD repeats are conserved regions, which may facilitate formation of heterotrimeric or multiprotein complexes</td>
</tr>
<tr class="even">
<td>ENSG00000089053</td>
<td>ANAPC5</td>
<td>Anaphase Promoting Complex Subunit 5</td>
</tr>
<tr class="odd">
<td>ENSG00000111361</td>
<td>EIF2B1</td>
<td>Eukaryotic Translation Initiation Factor 2B Subunit Alpha</td>
</tr>
<tr class="even">
<td>ENSG00000086475</td>
<td>SEPHS1</td>
<td>Selenophosphate Synthetase 1</td>
</tr>
<tr class="odd">
<td>ENSG00000106609</td>
<td>TMEM248</td>
<td>Transmembrane Protein 248</td>
</tr>
</tbody>
</table>
<p>All of the genes are related to conserved cellular functions - house keeping genes.</p>
</section>
</section>
</section>
<section id="using-the-cpm-values-answer-the-following-questions" class="level3" data-number="2.1.9">
<h3 data-number="2.1.9" class="anchored" data-anchor-id="using-the-cpm-values-answer-the-following-questions"><span class="header-section-number">2.1.9</span> Using the CPM values, answer the following questions:</h3>
<section id="which-5-genes-show-the-strongest-correlation-to-age-in-the-control-group" class="level4" data-number="2.1.9.1">
<h4 data-number="2.1.9.1" class="anchored" data-anchor-id="which-5-genes-show-the-strongest-correlation-to-age-in-the-control-group"><span class="header-section-number">2.1.9.1</span> Which 5 genes show the strongest correlation to age in the control group?</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the dataframe with the list of sample names with NF</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>NF_columns <span class="ot">&lt;-</span> <span class="fu">rownames_to_column</span>(sampleInfo, <span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">filter</span>(etiology <span class="sc">==</span> <span class="st">"NF"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">select</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>          <span class="fu">pull</span>(sample)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the gene expression data from the NF patients</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>NF_data <span class="ot">&lt;-</span> gxData <span class="sc">%&gt;%</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>           dplyr<span class="sc">::</span><span class="fu">select</span>(NF_columns)</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Transpose the dataframe to have genes as columns </span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>NF_data <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(NF_data))</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the metadata from NF patients</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>t_sampleInfo <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(sampleInfo))</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>NF_metadata <span class="ot">&lt;-</span> t_sampleInfo <span class="sc">%&gt;%</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>               <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"covariate"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>               dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(NF_columns, covariate)) <span class="sc">%&gt;%</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>               <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"covariate"</span>)</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Transpose the dataframe to have age as column</span></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>NF_metadata <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(NF_metadata))</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Add age column into the gene expression dataframe</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>NF_data <span class="ot">&lt;-</span> NF_data <span class="sc">%&gt;%</span></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>            <span class="fu">mutate</span>(<span class="at">age =</span> NF_metadata<span class="sc">$</span>age)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate correlation values</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>data_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(NF_data[ , <span class="fu">colnames</span>(NF_data) <span class="sc">!=</span> <span class="st">"age"</span>],</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>                <span class="fu">as.numeric</span>(NF_data<span class="sc">$</span>age))</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>data_cor <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(data_cor)</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Select the 5 most correlated genes with age in the control group</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>corr5_genes_age <span class="ot">&lt;-</span> data_cor <span class="sc">%&gt;%</span> </span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(V1))) <span class="sc">%&gt;%</span> </span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n=</span> <span class="dv">5</span>)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the gene expression data</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>corr5_gene_data <span class="ot">&lt;-</span> NF_data <span class="sc">%&gt;%</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>           dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">row.names</span>(corr5_genes_age))</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate significance (p value) of the 5 most correlated genes with age</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>corr5_genes_age[<span class="dv">1</span>,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(corr5_gene_data<span class="sc">$</span>ENSG00000244681, <span class="fu">as.numeric</span>(NF_data<span class="sc">$</span>age))<span class="sc">$</span>p.value</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>corr5_genes_age[<span class="dv">2</span>,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(corr5_gene_data<span class="sc">$</span>ENSG00000244694, <span class="fu">as.numeric</span>(NF_data<span class="sc">$</span>age))<span class="sc">$</span>p.value </span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>corr5_genes_age[<span class="dv">3</span>,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(corr5_gene_data<span class="sc">$</span>ENSG00000182264, <span class="fu">as.numeric</span>(NF_data<span class="sc">$</span>age))<span class="sc">$</span>p.value </span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>corr5_genes_age[<span class="dv">4</span>,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(corr5_gene_data<span class="sc">$</span>ENSG00000154080, <span class="fu">as.numeric</span>(NF_data<span class="sc">$</span>age))<span class="sc">$</span>p.value </span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>corr5_genes_age[<span class="dv">5</span>,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(corr5_gene_data<span class="sc">$</span>ENSG00000250337, <span class="fu">as.numeric</span>(NF_data<span class="sc">$</span>age))<span class="sc">$</span>p.value</span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(corr5_genes_age) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"estimate"</span>, <span class="st">"p_value"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li><strong>Is the correlation positive or negative?</strong></li>
</ul>
<p>4 of the values are positive and one is negative</p>
<ul>
<li><strong>Is the correlation significant?</strong></li>
</ul>
<p>Yes, all of the correlations were significant</p>
</section>
<section id="what-is-their-function-according-to-the-genecards-website-are-they-genes-of-which-the-expression-is-known-to-change-with-age-use-pubmed" class="level4" data-number="2.1.9.2">
<h4 data-number="2.1.9.2" class="anchored" data-anchor-id="what-is-their-function-according-to-the-genecards-website-are-they-genes-of-which-the-expression-is-known-to-change-with-age-use-pubmed"><span class="header-section-number">2.1.9.2</span> What is their function according to the GeneCards website? Are they genes of which the expression is known to change with age (use Pubmed)?</h4>
<table class="table">
<colgroup>
<col style="width: 33%">
<col style="width: 33%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th>Gene ID</th>
<th>Name</th>
<th>Function</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000244681</td>
<td>MTHFD2P1</td>
<td>Pseudogene</td>
</tr>
<tr class="even">
<td>ENSG00000244694</td>
<td>PTCHD4</td>
<td>Predicted to be integral component of membrane</td>
</tr>
<tr class="odd">
<td>ENSG00000182264</td>
<td>IZUMO1</td>
<td>The sperm-specific protein Izumo is essential for sperm-egg plasma membrane binding and fusion</td>
</tr>
<tr class="even">
<td>ENSG00000154080</td>
<td>CHST9</td>
<td>Catalyzes the transfer of sulfate to position 4 of non-reducing N-acetylgalactosamine (GalNAc) residues in both N-glycans and O-glycans</td>
</tr>
<tr class="odd">
<td>ENSG00000250337</td>
<td>PURPL</td>
<td>RNA Gene, and is affiliated with the lncRNA class. Diseases associated with colorectal cancer and myasthenic syndrome</td>
</tr>
</tbody>
</table>
</section>
<section id="visualize-the-result-for-at-least-1-gene-hint-cpm-values-on-the-y-axis-age-in-years-on-the-x-axis" class="level4" data-number="2.1.9.3">
<h4 data-number="2.1.9.3" class="anchored" data-anchor-id="visualize-the-result-for-at-least-1-gene-hint-cpm-values-on-the-y-axis-age-in-years-on-the-x-axis"><span class="header-section-number">2.1.9.3</span> Visualize the result for at least 1 gene (HINT: CPM values on the y-axis, age in years on the x-axis)</h4>
<p><strong>Scatter plot of one gene vs age</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add age column into the 5 most correlated genes dataframe</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>corr5_gene_data <span class="ot">&lt;-</span> corr5_gene_data <span class="sc">%&gt;%</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>            <span class="fu">mutate</span>(<span class="at">Age =</span> <span class="fu">as.numeric</span>(NF_metadata<span class="sc">$</span>age), <span class="at">.before =</span> <span class="dv">1</span>)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>corr1gene_age <span class="ot">&lt;-</span> corr5_gene_data <span class="sc">%&gt;%</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>        <span class="fu">select</span>(Age, ENSG00000244681)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>corr1gene_age_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(corr1gene_age, <span class="fu">aes</span>(<span class="at">x =</span> Age, <span class="at">y =</span> ENSG00000244681)) <span class="sc">+</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Age"</span>, <span class="at">y =</span> <span class="st">"ENSG00000244681 Expression"</span>) <span class="sc">+</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">n.breaks =</span> <span class="fl">10.0</span>)</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>corr1gene_age_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="MSB1005_SkillsTrainingWeek4_2022_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Scatter plots, correlation values, and distributions of all genes and age</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>corr_plot_allgenes <span class="ot">&lt;-</span> <span class="fu">ggcorrm</span>(<span class="at">data =</span> corr5_gene_data) <span class="sc">+</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_corrm</span>(<span class="at">base_size =</span> <span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">5</span>),</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">5</span>),</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>)) <span class="sc">+</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lotri</span>(<span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lotri</span>(<span class="fu">geom_smooth</span>(<span class="at">colour =</span> <span class="st">"red4"</span>)) <span class="sc">+</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">utri_heatmap</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">corr_method =</span> <span class="st">"spearman"</span>) <span class="sc">+</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">utri_corrtext</span>(<span class="at">corr_method =</span> <span class="st">"spearman"</span>, <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dia_histogram</span>(<span class="at">lower =</span> <span class="fl">0.1</span>, <span class="at">fill =</span> <span class="st">"grey80"</span>, <span class="at">color =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dia_density</span>(<span class="at">lower =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> .<span class="dv">1</span>, <span class="at">colour =</span> <span class="st">"red4"</span>) <span class="sc">+</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">"white"</span>, <span class="at">mid =</span> <span class="st">"red3"</span>, <span class="at">high =</span> <span class="st">"red4"</span>,</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>                       <span class="at">midpoint =</span> <span class="fl">0.5</span>, <span class="at">space =</span> <span class="st">"rgb"</span>,</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>                       <span class="at">guide =</span> <span class="fu">guide_colorbar</span>(<span class="at">title =</span> <span class="st">"Correlation coefficient"</span>),</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>                       <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>))</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>corr_plot_allgenes</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="MSB1005_SkillsTrainingWeek4_2022_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
</section>
<section id="assignment-4-differential-gene-expression-analysis." class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="assignment-4-differential-gene-expression-analysis."><span class="header-section-number">2.2</span> Assignment 4: Differential gene expression analysis.</h2>
<p>Now that we have explored the gene expression data, it is time to perform a differential gene expression analysis.</p>
<section id="what-is-differential-gene-expression-analysis-dgea-what-are-some-of-the-most-common-packages-in-r-for-dgea" class="level3" data-number="2.2.1">
<h3 data-number="2.2.1" class="anchored" data-anchor-id="what-is-differential-gene-expression-analysis-dgea-what-are-some-of-the-most-common-packages-in-r-for-dgea"><span class="header-section-number">2.2.1</span> What is differential gene expression analysis (DGEA)? What are some of the most common packages in R for DGEA?</h3>
<p>Differential expression analysis means taking the normalised read count data and performing statistical analysis to discover quantitative changes in<br>
expression levels between experimental groups.</p>
<p><strong>R packages:</strong></p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/limma.html">limma</a></li>
</ul>
<p>We are going to use the limma package to perform a DGEA. We need to use the CPM normalized values. Have a look at the limma guide section 15.4: (https://www.bioconductor.org/packages/devel/bioc/vignettes/limma/inst/doc/usersguide.pdf)</p>
</section>
<section id="implement-the-steps-noted-in-the-limma-guide-for-the-magnet-dataset.-start-with-a-dgea-between-dcm-patients-and-healthy-controls." class="level3" data-number="2.2.2">
<h3 data-number="2.2.2" class="anchored" data-anchor-id="implement-the-steps-noted-in-the-limma-guide-for-the-magnet-dataset.-start-with-a-dgea-between-dcm-patients-and-healthy-controls."><span class="header-section-number">2.2.2</span> Implement the steps noted in the limma guide for the MAGNET dataset. Start with a DGEA between DCM patients and healthy controls.</h3>
<section id="limma-trend" class="level4" data-number="2.2.2.1">
<h4 data-number="2.2.2.1" class="anchored" data-anchor-id="limma-trend"><span class="header-section-number">2.2.2.1</span> Limma-trend</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert counts to logCPM values</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>logCPM <span class="ot">&lt;-</span> <span class="fu">cpm</span>(gxData, <span class="at">log =</span> <span class="cn">TRUE</span>, <span class="at">prior.count =</span> <span class="dv">3</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Create design matrix</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>design <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> sampleInfo<span class="sc">$</span>etiology)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply limma pipeline</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(logCPM, design)</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit, <span class="at">trend =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="fu">topTable</span>(fit, <span class="at">coef =</span> <span class="fu">ncol</span>(design))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   logFC  AveExpr        t P.Value adj.P.Val        B
ENSG00000089053 6.381604 6.384943 961.3848       0         0 1415.207
ENSG00000129351 6.429434 6.416465 955.8661       0         0 1413.270
ENSG00000105323 6.380228 6.365844 944.4249       0         0 1409.213
ENSG00000136709 6.329745 6.324768 937.9541       0         0 1406.892
ENSG00000106609 6.348231 6.340574 921.1237       0         0 1400.769
ENSG00000176915 6.281051 6.282988 907.5412       0         0 1395.732
ENSG00000075785 6.481443 6.478184 898.0510       0         0 1392.161
ENSG00000100711 6.284988 6.278731 891.0022       0         0 1389.480
ENSG00000182944 6.459927 6.446393 890.8330       0         0 1389.416
ENSG00000113648 6.404155 6.400461 890.3165       0         0 1389.218</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Give more weight to fold-changes in the gene ranking</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(logCPM, design)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">treat</span>(fit, <span class="at">lfc =</span> <span class="fu">log2</span>(<span class="fl">1.2</span>), <span class="at">trend =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="fu">topTreat</span>(fit, <span class="at">coef =</span> <span class="fu">ncol</span>(design))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   logFC  AveExpr        t P.Value adj.P.Val
ENSG00000000003 6.117085 6.111910 388.5360       0         0
ENSG00000000419 6.294569 6.292874 480.5645       0         0
ENSG00000000457 6.100006 6.111824 433.3020       0         0
ENSG00000000460 5.886348 5.887347 308.1765       0         0
ENSG00000000938 5.992674 6.059126 206.5199       0         0
ENSG00000000971 6.391421 6.392179 255.8019       0         0
ENSG00000001036 6.299640 6.308049 581.1499       0         0
ENSG00000001084 6.273295 6.255830 284.7585       0         0
ENSG00000001167 6.134147 6.122626 201.0493       0         0
ENSG00000001460 5.943226 5.954836 370.8629       0         0</code></pre>
</div>
</div>
<p>Results show the p value of 0 in some genes, which means that something is wrong.</p>
</section>
</section>
<section id="which-co-variates-should-be-taken-along-for-correction-confounding-see-the-alcohol-causes-lung-cancer-example-from-the-lecture" class="level3" data-number="2.2.3">
<h3 data-number="2.2.3" class="anchored" data-anchor-id="which-co-variates-should-be-taken-along-for-correction-confounding-see-the-alcohol-causes-lung-cancer-example-from-the-lecture"><span class="header-section-number">2.2.3</span> Which co-variates should be taken along for correction? (confounding; see the “alcohol causes lung cancer” example from the lecture)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert counts to logCPM values</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>logCPM <span class="ot">&lt;-</span> <span class="fu">cpm</span>(gxData, <span class="at">log =</span> <span class="cn">TRUE</span>, <span class="at">prior.count =</span> <span class="dv">3</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Create design matrix considering confounding variables</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>design <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> etiology <span class="sc">+</span> gender <span class="sc">+</span> age, <span class="at">data =</span> sampleInfo)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply limma pipeline with confounding variables</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(logCPM, design)</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>cont.matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">DCMvsControl =</span> etiologyDCM <span class="sc">-</span> etiologyNF,</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>                             <span class="at">levels =</span> design)</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit, cont.matrix)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>efit <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit, <span class="at">trend =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>dgeRes <span class="ot">&lt;-</span> <span class="fu">topTable</span>(efit, <span class="at">coef =</span> <span class="st">'DCMvsControl'</span>, <span class="at">number =</span> <span class="fu">nrow</span>(gxData))</span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(dgeRes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Rows: 20,781
Columns: 6
$ logFC     &lt;dbl&gt; 0.15831103, 0.33966098, 0.21263013, -0.44141913, 0.36628640,…
$ AveExpr   &lt;dbl&gt; 6.286287, 5.952977, 6.101884, 6.022681, 6.049712, 6.371697, …
$ t         &lt;dbl&gt; 29.01545, 27.11832, 25.18996, -25.06301, 24.77563, -24.26241…
$ P.Value   &lt;dbl&gt; 7.851116e-97, 1.689807e-89, 7.264246e-82, 2.342191e-81, 3.33…
$ adj.P.Val &lt;dbl&gt; 1.631540e-92, 1.755794e-85, 5.031943e-78, 1.216827e-77, 1.38…
$ B         &lt;dbl&gt; 210.1073, 193.3613, 175.9173, 174.7550, 172.1180, 167.3887, …</code></pre>
</div>
</div>
<p>After including the cofounding variables, the p values are not 0, which means that the cofounding variables are important for the analysis.</p>
</section>
<section id="copy-the-top-200-differentially-expressed-genes-to-for-a-quick-go-enrichment-analysis.-which-processes-are-changed-between-dcm-and-controls-do-these-processes-make-biological-sense-quick-literature-check" class="level3" data-number="2.2.4">
<h3 data-number="2.2.4" class="anchored" data-anchor-id="copy-the-top-200-differentially-expressed-genes-to-for-a-quick-go-enrichment-analysis.-which-processes-are-changed-between-dcm-and-controls-do-these-processes-make-biological-sense-quick-literature-check"><span class="header-section-number">2.2.4</span> Copy the top 200 differentially expressed genes to for a quick <a href="http://cbl-gorilla.cs.technion.ac.il/">GO enrichment analysis</a>. Which processes are changed between DCM and controls? Do these processes make biological sense? (quick literature check!)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select the 200 most correlated genes with age in the control group</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>to200_corr_genes <span class="ot">&lt;-</span> dgeRes <span class="sc">%&gt;%</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">200</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"Gene_ID"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Gene_ID)</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Select the names of all genes in the control group</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>all_corr_genes <span class="ot">&lt;-</span> dgeRes <span class="sc">%&gt;%</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"Gene_ID"</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Gene_ID)</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Export target list to csv file</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(to200_corr_genes, <span class="st">"Outputs/to200_corr_genes.csv"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Export background list to csv file</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(all_corr_genes, <span class="st">"Outputs/all_corr_genes.csv"</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>In the <a href="http://cbl-gorilla.cs.technion.ac.il/">GOrilla server</a>, the inputs are the target and background tables exported in the previous step. The results are shown in the following figure:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Outputs/gorilla_results.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">GOrilla results</figcaption><p></p>
</figure>
</div>
<p>The results show that the most enriched GO terms are related to inflammation and structural processed, and the immune system, which is consistent with the literature.</p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./MSB1005_SkillsTrainingWeek3_2022.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Week3 - Introduction to Gene expression analysis part 1</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav>
</div> <!-- /content -->



</body></html>